### DISCUSSION TITLE
Authentication Error Help

### DISCUSSION THREAD

**Discussion by pkillionFlora** [saleweaver/python-amazon-sp-api#322] [Q&A] [answered]
Hi guys, 

I am attempting to utilize the getorders and create report endpoint and every endpoint I try I get the error:

sp_api.auth.exceptions.AuthorizationError: ('invalid_client', 'Clie\site-packages\sp_api\auth\access_token_client.py", line 36, in _requestnt authentication failed', 401)   

Further up in the exception it says: KeyError: 'access_token_*******************' 

I saw some other posts mentioning redoing the mws IAM, and I did but I did not come across any glaring errors, any tips?

**Comment by saleweaver** [comment]
Hi,

please try to remove the role_arn from the credentials dict.

**Reply by pkillionFlora** [reply]
I did that and I am receiving the same error.

**Comment by saleweaver** [comment, marked as answer]
I've just tried a couple of combinations. The IAM part seems to be setup fine (not sure, but it seems that's working for you). 

Try to make sure that the following values are valid and have the correct association with the role/user in AWS:

refresh_token
lwa_app_id
lwa_client_secret

All those settings have to be done in SellerCentral

**Reply by pkillionFlora** [reply]
Hi I double checked those settings but am seeing the same error code. I have noticed that my LWA_APP_ID is amzn1.sp.solution.************* while others start with LWA_APP_ID=amzn1.application-oa2-client.8c42****** is this because I did not set up Oauth 2? Is oauth 2 necessary?

Edit: solved was using the wrong app_id

**Reply by rashedoz** [reply]
@pkillionFlora Thanks for the Edit update. I was also using the wrong APP ID instead of the LWA_APP_ID.
